homeassistant/components/fitbit/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow for fitbit."""
+
+import logging
+from typing import Any
+
+from fitbit.exceptions import HTTPException
+
+from homeassistant.const import CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResult, FlowResultType
+from homeassistant.helpers import config_entry_oauth2_flow
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from . import api
+from .const import DOMAIN, OAUTH_SCOPES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Config flow to handle fitbit OAuth2 authentication."""
+
+    DOMAIN = DOMAIN
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    @property
+    def extra_authorize_data(self) -> dict[str, str]:
+        """Extra data that needs to be appended to the authorize url."""
+        return {
+            "scope": " ".join(OAUTH_SCOPES),
+            "prompt": "consent",
+        }
+
+    async def async_oauth_create_entry(self, data: dict[str, Any]) -> FlowResult:
+        """Create an entry for the flow, or update existing entry."""
+
+        client = api.ConfigFlowFitbitApi(self.hass, data[CONF_TOKEN])
+        try:
+            profile = await client.async_get_user_profile()
+        except HTTPException as err:
+            _LOGGER.error("Failed to fetch user profile for Fitbit API: %s", err)
+            return self.async_abort(reason="cannot_connect")
+
+        await self.async_set_unique_id(profile.encoded_id)
+        self._abort_if_unique_id_configured()
+        return self.async_create_entry(title=profile.full_name, data=data)
+
+    async def async_step_import(self, data: dict[str, Any]) -> FlowResult:
+        """Handle import from YAML."""
+        result = await self.async_oauth_create_entry(data)
+        if result.get("type") == FlowResultType.ABORT:
+            async_create_issue(
+                self.hass,
+                DOMAIN,
+                "deprecated_yaml_import_issue_cannot_connect",
```

Comment:
I don't think we want to create multiple issues with different institutions at the same time. I think we should use the same issue key and different translation keys so we can overwrite the issue, or make sure we only register one issue at a time.

Reply:
Fixed to use the same issue key for all imports, and updated the tests to ensure there is only one issue, but with different translation keys.

This also makes me think that the logic for import and creating issues should be not spread across two files -- either the platform setup parses the config flow response (instead of creating the issue in the config flow).  Or alternatively, move all of the import logic into the config flow -- however that would mean running import even when we know the config can't be imported in the end because its missing fields. Just a thought, but leaving code as is, but if you think there is something there i can do one of those.

Reply:
We could await the config flow in the platform and check the result.

Reply:
OK, that is kind of what i had in mind, though its rare we see it done that way it seems like it keeps this simpler overall.  I'll give this a try.

Reply:
OK, i think i like this better for simplifying repair issue creation and separating the concerns, and I feel fairly good about the existing test coverage of a few types of import abort cases exercising this.